### PR TITLE
Document enhancement issues

### PR DESCRIPTION
**Pull request by danehans**
This PR adds docs for proposing enhancements to service-apis.

__Note:__ Enhancements include a User Stories section, so [this gdoc](https://docs.google.com/document/d/1OAzVO0HfmF4qD8AW41_eeUY0KP6xjqghusUVwhp4BNg/edit?usp=sharing) should be deprecated.

/assign @bowei 

cc: @ironcladlou @Miciah

### CODE REVIEW

**Comment by bowei** [comment]
I think this totally makes sense when we have a one (or more) alpha versions published. 
It may be heavy weight right now given the amount of discussion in defining the API. 
It would be useful to try to use the meeting discussions and Github issues for the time being until the group is generally comfortable for the big pieces.

**Comment by bowei** [comment]
This would make a great issue template -- I'll create one.

**Comment by danehans** [comment]
@bowei commit `a1721c0` addresses your feedback in https://github.com/kubernetes-sigs/service-apis/pull/46#issuecomment-576557438 by greatly simplifying the enhancement process.

**Comment by danehans** [comment]
> This would make a great issue template -- I'll create one.

@bowei I updated the template in commit `a1721c0`, ptal. Also note that the enhance quick start states:

> Create an [Issue](https://github.com/kubernetes-sigs/service-apis/issues/new/choose) and select "Enhancement Request".

I will update this PR accordingly after you create the enhancement request template.

**Comment by danehans** [comment]
@bowei I looked into GH [issue templates](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository) and updated the PR to support an Enhancement Request template.

File: docs-src/enhancement-requests.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Create an [Issue](https://github.com/kubernetes-sigs/service-apis/issues/new/choose) and select
+"Enhancement Request".
+2. Follow the instructions in the enhancement request template and submit the Issue.
+
+## What is Considered an Enhancement?
+
+An enhancement is generally anything that:
+
+- impacts how a cluster is operated including addition or removal of significant
```
Comment (bowei) [inline docs-src/enhancement-requests.md]: probably want to change this to say 

- impacts the service apis, including addition or removal... etc

proposal lgtm, let's discuss in meeting today to give everyone a chance comment, then we can merge.

**Comment by bowei** [comment]
looks like there are some conflicts...

**Pull request merged**